Sources/Jobs/Scheduler.swift

```diff
@@ -108,11 +113,11 @@ private extension Scheduler {
     }
     
     func subscribe(job: Job, to observedObject: AnyObservedObject) {
-        observedObject.valueDidChange = {
-            observedObject.setChanged(to: true)
+        self.observations.append(observedObject.register {
+            observedObject.changed = true
             job.run()
-            observedObject.setChanged(to: false)
-        }
+            observedObject.changed = false
+        })
```

Comment:
I don't think that it is necessary to store all observations in the `Scheduler`

Reply:
Yes, it is necessary. Otherwise the `Observation` isn't kept in memory and the registered callback is lost. You can observe this by running the tests and placing a breakpoint on `job.run()`. It won't be executed when the `observations.append` was removed. For some reason your tests still pass if removing, so the tests aren't strict enough.

Reply:
I modified on job-test so it actually checks if the job is run in 8631127.